instrument/invariant_test.go

```diff
@@ -38,3 +41,47 @@ func ExampleInvariantViolatedMetricInvocation() {
 	fmt.Println(counter.Name(), counter.Value())
 	// Output: invariant_violated 1
 }
+
+func TestEmitInvariantViolationDoesNotPanicIfEnvNotSet(t *testing.T) {
+	opts := instrument.NewOptions()
+	instrument.EmitInvariantViolation(opts)
+}
+
+func TestEmitAndLogInvariantViolationDoesNotPanicIfEnvNotSet(t *testing.T) {
+	opts := instrument.NewOptions()
+	instrument.EmitAndLogInvariantViolation(opts, func(l log.Logger) {
+		l.Error("some_error")
+	})
+}
+
+func TestEmitInvariantViolationPanicsIfEnvSet(t *testing.T) {
+	os.Setenv(instrument.ShouldPanicEnvironmentVariableName, "true")
+	defer func() {
+		if r := recover(); r == nil {
+			panic("test should have panic'd but did not")
+		}
+	}()
```

Comment:
Instead of this defer, can you use:
```go
import

func TestMyPanic(t *testing.T) {
  assert.Panics(t, func() {
    // ...
  })
}

func TestMyNotPanic(t *testing.T) {
  assert.NotPanics(t, func() {
    // ...
  })
}
```

Also for the set and defer set env, can we DRY this just a tad:
```go
type cleanupFn func()

func setShouldPanicEnvironmentVariable() cleanupFn {
  restoreValue := os.Getenv(instrument.ShouldPanicEnvironmentVariableName)
  os.Setenv(instrument.ShouldPanicEnvironmentVariableName, "true")
  return cleanupFn(func() {
    os.Setenv(instrument.ShouldPanicEnvironmentVariableName, restoreValue)
  })
}

func TestMy(t *testing.T) {
  defer setShouldPanicEnvironmentVariable()()
  assert.Panics(t, func() {
    
  })
}
```